src/App.tsx

```diff
@@ -1,51 +1,100 @@
-      <header className="App__header">
-        <h1 className="App__title">
-          Jam is selected
-        </h1>
-
-        <button
-          type="button"
-          className="App__clear"
-        >
-          Clear
-        </button>
-      </header>
-
-      <ul>
-        <li className="Good">Dumplings</li>
-        <li className="Good">Carrot</li>
-        <li className="Good">Eggs</li>
-        <li className="Good">Ice cream</li>
-        <li className="Good">Apple</li>
-        <li className="Good">Bread</li>
-        <li className="Good">Fish</li>
-        <li className="Good">Honey</li>
-        <li className="Good Good--active">Jam</li>
-        <li className="Good">Garlic</li>
-      </ul>
-
-      {/* Put required buttons into each Good */}
-      <button
-        type="button"
-        className="Good__remove"
-      >
-        Remove
-      </button>
-
-      <button
-        type="button"
-        className="Good__select"
-      >
-        Select
-      </button>
-    </main>
-  );
+import goodsFromServer from './goods';
+
+type State = {
+  selectedGoods: string[];
 };
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: ['Jam'],
+  };
+
+  clearProducts = () => {
+    this.setState({ selectedGoods: [] });
+  };
+
+  selectProduct = (product: string) => {
+    this.setState((state) => ({
+      selectedGoods: [...state.selectedGoods, product],
+    }));
+  };
+
+  removeProduct = (product: string) => {
+    this.setState((state) => ({
+      selectedGoods: state.selectedGoods.filter(item => item !== product),
+    }));
+  };
+
+  changeButton = (isSelected: boolean, product: string) => {
+    return isSelected
+      ? this.removeProduct(product)
+      : this.selectProduct(product);
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+    const showTitle = (product: string[]): string => {
+      if (product.length === 0) {
+        return ' No goods selected';
+      }
+
+      if (product.length === 1) {
+        return ` ${product[0]} is selected`;
+      }
+
+      return `${product.slice(0, -1).join(', ')} and ${product.slice(-1)} are selected`;
+    };
+
+    return (
+      <main className="App">
+        <header className="App__header">
+          <h1 className="App__title">
+            {showTitle(selectedGoods)}
+          </h1>
+        </header>
+
+        <ul className="App__goods">
+          {goodsFromServer.map(product => {
+            const isSelected = selectedGoods.includes(product);
+            const buttonName = isSelected ? 'Remove' : 'Select';
+
+            return (
+              <li
+                className="App__product"
+                key={product}
+              >
+                {product}
+                <button
+                  className={
+                    !isSelected
+                      ? 'App__button'
+                      : 'App__button App__button--active'
+                  }
+                  type="button"
```

Comment:
this will look better in a separate component

Reply:
I agree with you, but we shouldn't create additional components it's [one of conditions](https://prnt.sc/eCr6lWLI2-tw)